mitodl/mit-open
# dashboard menu

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4456

### Description (What does it do?)
This PR adds the dashboard menu for both desktop and mobile designs as shown in the issue above. Placeholder content was put in place for the "Home" and "Profile" sections, and the current User List UI was put into the "My Lists" section. Later work will introduce the actual Home and Profile sections and overhaul the My Lists section.

### Screenshots (if appropriate):
<img width="960" alt="chrome_kQT6fOa36X" src="https://github.com/mitodl/mit-open/assets/12089658/ed17ff3f-4418-415d-b6b9-3c3a36bbc39d">
<img width="189" alt="chrome_tCtbH36aWU" src="https://github.com/mitodl/mit-open/assets/12089658/df756cae-427c-49ee-b5aa-be3a9aecd47c">
<img width="960" alt="chrome_3oAV7ChYpp" src="https://github.com/mitodl/mit-open/assets/12089658/faeb3efc-57a0-4e83-a6ca-1953fbf83b0a">
<img width="189" alt="chrome_AA8xNR3QWw" src="https://github.com/mitodl/mit-open/assets/12089658/3f22a708-1d71-4507-aba1-f511dad4cc40">

### How can this be tested?
 - Spin up `mit-open` on this branch
 - Log in to your local instance of `mit-open` at http://localhost:8063/
 - Click the header menu in the upper right and click "Dashboard"
 - Click through the various dashboard tabs and ensure the content loads correctly. You should see placeholder pages for "Home" and "Profile," but the "My Lists" tab should load the current User List UI.
 - Test the My Lists UI:
   - Click "Create new list" and fill out the modal form
   - Click the title of the list you just created (it should be empty)
   - Open a new tab and go to http://localhost:8063/ (the home page)
   - On a few of the courses listed there in the carousel, click the bookmark icon and select the list you just created
   - Switch back to the dashboard tab and confirm that the items you added now show up in the list
   - Click the Reorder button and move some of the courses around in the list
   - Click "done ordering" and verify that reordering was successful
   - Click "edit" and change the title of your list, then hit save
   - Verify that the title changes on the page
   - Click the "My Lists" menu button to go back to the "list of lists"
 - Repeat all of this at mobile resolutions
